### PR TITLE
Build: Use jom to parallelize Windows builds

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**

This PR installs [Qt jom](https://wiki.qt.io/Jom) via `choco` and uses it in place of `nmake`. `jom` supports parallel builds and this PR enables them.

CHANGELOG: Internal: Speed up build process by using jom with parallelization instead of nmake in autobuilds.

**Context: Fixes an issue?**

Performance

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

- [x] Windows build times are down from [8m52s](https://github.com/jamulussoftware/jamulus/runs/5366322321?check_suite_focus=true) on master to [6m16s](https://github.com/hoffie/jamulus/runs/5383808077?check_suite_focus=true) in this build [[5m54s in the most recent run]](https://github.com/jamulussoftware/jamulus/runs/5387615640?check_suite_focus=true) (only compared JACK, but ASIO should be similar)

Note: Build times do vary so take these numbers with a grain of salt.

**What is missing until this pull request can be merged?**

CI/Reviews

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I tested my code and it does what I want
- [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I've filled all the content above

### CODE REVIEW

File: windows/deploy_windows.ps1 [hunk 1]
```diff
@@ -234,7 +234,15 @@ Function Build-App
...
     Set-Location -Path $BuildPath
-    Invoke-Native-Command -Command "nmake" -Arguments ("$BuildConfig")
+    if (Get-Command "jom.exe" -ErrorAction SilentlyContinue)
+    {
+        echo "Building with jom /J ${Env:NUMBER_OF_PROCESSORS}"
+        Invoke-Native-Command -Command "jom" -Arguments ("/J", "${Env:NUMBER_OF_PROCESSORS}", "$BuildConfig")
+    }
+    else
+    {
+        Invoke-Native-Command -Command "nmake" -Arguments ("$BuildConfig")
```
Comment (ann0see) [inline windows/deploy_windows.ps1:244]: I‘d give a message saying that jom couldn’t be found so the build is now done with nmake which might be slower.

**Comment by hoffie** [comment]
Quick test from a branch which has the `deploy_windows.ps1` changes from this PR, but does not install `jom` in the prepare script: https://github.com/hoffie/jamulus/runs/5387646863?check_suite_focus=true#step:9:126

(This was before @ann0see's suggestion to add an explicit echo for this case, so don't search for it ;))

- [x] Build logic still works even without `jom` installed

**Review by ann0see — APPROVED** [review]
Ok. So approving in these terms.

**Comment by ann0see** [comment]
In Linux I thought multiple files were compiled in parallel (at least locally) and I don’t know if you can judge via CI log.

**Comment by hoffie** [comment]
x

> In Linux I thought multiple files were compiled in parallel (at least locally) and I don’t know if you can judge via CI log.

There are two places where "parallelization" can happen:
- By running compiler invocations in parallel (that's what `make -j` and `jom /N` do. That's not necessarily visible in build logs, yep.
- By compiling multiple source files in a single compiler invocation. This isn't parallelization, but it'll likely reduce overall compiler warm up times, `.h` parsing, etc. This should be visible in the build logs as the invocation would have to change for that. That's where `/MP` would be relevant. But if we don't expect benefits, I'm a bit hesistant to add it.

File: autobuild/windows/autobuild_windowsinstaller_1_prepare.ps1 [hunk 2]
```diff
@@ -46,6 +46,7 @@ $AqtinstallVersion = "2.0.5"
 $JackVersion = "1.9.17"
 $Msvc32Version = "win32_msvc2019"
 $Msvc64Version = "win64_msvc2019_64"
+$JomVersion = "1.1.2"
```
Comment (pljones) [inline autobuild/windows/autobuild_windowsinstaller_1_prepare.ps1:49]: Do we have a documented process in place to review pinned versions from time to time?

(Don't I love coming up with unconnected issues...)

Comment (hoffie) [reply autobuild/windows/autobuild_windowsinstaller_1_prepare.ps1:49]: > Do we have a documented process in place to review pinned versions from time to time?

Not yet, but there's at least *one* reminder: #2346

**Pull request merged**